src/blockencodings.cpp

```diff
@@ -130,6 +131,35 @@ ReadStatus PartiallyDownloadedBlock::InitData(const CBlockHeaderAndShortTxIDs& c
         if (mempool_count == shorttxids.size())
             break;
     }
+    }
```

Comment:
Seems I was misreading the code, and interpreting the early-exit `mempool_count == shorttxids.size()` as the way the loop always successfully ends.